packages/cozy-harvest-lib/src/components/KonnectorAccounts.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { withMutations } from 'cozy-client'
+import { withRouter } from 'react-router'
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
+import { ModalContent } from 'cozy-ui/transpiled/react/Modal'
+import { translate } from 'cozy-ui/transpiled/react/I18n'
+import Infos from 'cozy-ui/transpiled/react/Infos'
+import Button from 'cozy-ui/transpiled/react/Button'
+import get from 'lodash/get'
+
+import accountMutations from '../connections/accounts'
+import triggersMutations from '../connections/triggers'
+import * as triggersModel from '../helpers/triggers'
+import KonnectorModalHeader from './KonnectorModalHeader'
+
+class KonnectorAccounts extends React.Component {
+  state = {
+    fetchingAccounts: false,
+    error: null,
+    accounts: []
+  }
+
+  /**
+   * TODO We should not fetchAccounts and fetchAccount since we already have the informations
+   * in the props. We kept this sytem for compatibility on the existing override.
+   * Next tasks: remove these methods and rewrite the override
+   */
+  async componentDidMount() {
+    await this.fetchAccounts()
+  }
+
+  componentDidUpdate(prevProps) {
+    // After leaving the new account screen, we need to refetch accounts in case a new one was created
+    // TODO: We should use an AccountsWatcher (realtime)
+    if (
+      this.props.location.pathname !== prevProps.location.pathname &&
+      /\/new\/?$/.test(prevProps.location.pathname)
+    ) {
+      this.fetchAccounts()
+    }
+  }
+
+  async fetchAccounts() {
+    const triggers = get(this.props, 'konnector.triggers.data', [])
+    const { findAccount } = this.props
+    this.setState({ fetchingAccounts: true })
+    try {
+      const accounts = (await Promise.all(
+        triggers.map(async trigger => {
+          return {
+            account: await findAccount(triggersModel.getAccountId(trigger)),
+            trigger
+          }
+        })
+      )).filter(({ account }) => !!account)
+      this.setState({ accounts, error: null })
+    } catch (error) {
+      this.setState({ error })
+    } finally {
+      this.setState({ fetchingAccounts: false })
+    }
+  }
+
+  render() {
+    const { accounts, fetchingAccounts, error } = this.state
+    const { konnector, t } = this.props
+    return (
+      <>
```

Comment:
It is surprising to me to have such a "complex" render logic here. When seeing the usage of KonnectorAccounts, I thought it would only handle data fetching. Consider splitting the component or finding another name ?